### PR TITLE
TK - database update to add refers

### PR DESCRIPTION
**Pull request by tskeeley**
This adds in all the referential info we've been talking about, needed it to setup for the neon imports

### CODE REVIEW

**Comment by AricStewart** [comment]
[Fri Mar 02 09:46:56.071856 2018] [php7:error] [pid 314] [client ::1:53278] PHP Fatal error:  Uncaught PDOException: SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (`con_concomsi`.`#sql-480_10`, CONSTRAINT `#sql-480_10_ibfk_1` FOREIGN KEY (`RewardGroupID`) REFERENCES `RewardGroup` (`RewardGroupID`) ON UPDATE CASCADE) in /Users/aric/Personal/CON/CIAB-Signin/functions/database.inc:88\nStack trace:\n#0 /Users/aric/Personal/CON/CIAB-Signin/functions/database.inc(88): PDO->query('ALTER TABLE Vol...')\n#1 /Users/aric/Personal/CON/CIAB-Signin/functions/update/from_152_to_153.inc(67): DB::run('ALTER TABLE Vol...')\n#2 /Users/aric/Personal/CON/CIAB-Signin/functions/update/from_152_to_153.inc(75): _updatetables153_DBSchema()\n#3 /Users/aric/Personal/CON/CIAB-Signin/functions/update.inc(34): from_152_to_153()\n#4 /Users/aric/Personal/CON/CIAB-Signin/pages/body/update.inc(18): db_do_update('2018011700')\n#5 /Users/aric/Personal/CON/CIAB-Signin/index.php(72): require('/Users/aric/Per...')\n#6 {main}\n  thrown in /Users/aric/Personal/CON/CIAB-Signin/functions/database.inc on line 88

File: pages/body/update.inc [hunk 1]
```diff
@@ -10,9 +13,6 @@ if (!empty($_REQUEST)) {
     ];
 
     $updateData = filter_input_array(INPUT_GET, $arguments);
-    echo "<div>\n";
-    echo "<button type='button' class='w3-button w3-green w3-left' onclick=window.location.href='http://".$_SERVER['SERVER_NAME']."/index.php?Function=".$updateData['next']."'> Proceed to ".$updateData['next']."</button>\n";
-    echo "</div>\n";
```
[the remark this answers is not in the document]
Comment (tskeeley) [reply pages/body/update.inc]: I removed the first one, that shows up before the work is complete.  I clicked it once and jumped off the page before updates where done, and it started a second run of updates.  which of course all went weird.  yikes!

File: index.php [hunk 2]
```diff
@@ -2,21 +2,25 @@
...
     if ($_REQUEST['Function'] != 'functions' && !strstr($_REQUEST['Function'], $_SESSION['kioskMode'])) {
         goSite('/index.php?Function='.$_SESSION['kioskMode']);
     }
 }
-// Divert to public page if no session is available
-if (empty($_REQUEST['Function'])) {
-    goSite('/index.php?Function=public');
-} elseif (!empty($_REQUEST['DeepLink'])) {
+
+if (!empty($_REQUEST['DeepLink'])) {
```
[the remark this answers is not in the document]
Comment (tskeeley) [reply index.php]: Deeplink is how to get a function without logging in.  Notably, the concom maillist output.  It was also used (err, was going to be used) to do confirmations in the hotel module but that was pulled.  There is a VERY back burner idea for allowing "approvals" which would be unique id deeplinks.  (EG: approve a subhead, approve update to the years worked, if we add a finance - approve expenses)

The order was moved around to solve some process repetition that came into existence with the module work and I ran into during the update.

Still have to solve the error, but that's separate.

**Comment by tskeeley** [comment]
The error is weird.  Adding in the refers constraints are breaking BEFORE we even try code?  hmmm  Canceling the pull until I can resolve  (and trying to find why it didn't happen for me)

File: pages/body/update.inc (hunk 1 above)
[the remark this answers is not in the document]
Comment (AricStewart) [reply pages/body/update.inc]: Understood.

File: index.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (AricStewart) [reply index.php]: ok makes sense

**Comment by AricStewart** [comment]
I found my error!

I had a reward group id that was not in the reward group table! 

Fixing that, now it updated cleanly! ALL MY ERROR!

**Comment by tskeeley** [comment]
Oh, well, then, Thank goodness! :)  I don't have to shread my code! :)

**Pull request merged**